### DISCUSSION TITLE
Svg generator for steno outlines

### DISCUSSION THREAD

**Discussion by IllustratedMan-code** [aerickt/lapwing-for-beginners#34] [Ideas]
Hi! I saw in the book that you would like help to shift all the outlines to SVG. I don't think it would be that difficult for me to create some custom javascript code to convert strokes (e.g. `SKR`) into svg images. It would even be possible to have a custom HTML element that would be specifiable by something like `<outline stroke="SKR">`.  It looks like you can include custom javascript into mdbook so it should be possible to have the custom elements available in markdown.  

I would only need the SVG that is used to generate the previews in the book to get started.

**Comment by aerickt** [comment]
Hello! Thank you so much for offering your help! The custom HTML idea is very awesome. Here are the SVG files:
![en_layout_top_s_num_blueberry_dark](https://github.com/user-attachments/assets/28970765-3603-4148-abf5-dd2e57b415be)
![en_layout_top_s_num_blueberry_light](https://github.com/user-attachments/assets/ef25f8c2-6710-47e8-aef0-23193985e9b6)

Let me know if there's anything else I can help you with.

**Reply by IllustratedMan-code** [reply]
Looks like I can't download these images because they are hosted in a private repository. I think you need to download the images and upload them here manually for me to access them.

Edit: Never mind, github was just being weird I think

**Comment by IllustratedMan-code** [comment]
I have a proof of concept here. I'll work on integrating it into this repo soon and I'll make a pull request.

Github Repo: https://github.com/IllustratedMan-code/steno-visualizer

Test website: https://illustratedman-code.github.io/steno-visualizer/

**Reply by aerickt** [reply]
Looks awesome! I have just a few minor suggestions.
The two files I sent are actually two separate colour schemes; the top is for the site's dark themes and the bottom is for the light ones. I was thinking that it could look a little like the plover-svg-layout-display plugin where unpressed keys are just blank.

So, `H-L/WORLD` would look something like this for the two themes:

![Screenshot From 2024-11-03 20-19-44](https://github.com/user-attachments/assets/16222467-a6ab-44b4-96fb-4cc1d1cd1b58)
![Screenshot From 2024-11-03 20-19-48](https://github.com/user-attachments/assets/8e4d1fef-2f09-40e2-b772-93a085009dfe)

![Screenshot From 2024-11-03 20-19-34](https://github.com/user-attachments/assets/d7f5908c-6e15-4256-a32b-1a6aa16c9af4)
![Screenshot From 2024-11-03 20-19-37](https://github.com/user-attachments/assets/a973acca-ab13-43e9-ade9-6ac1b5fd0ef8)

I'd also been looking at some of the default themes in mdbook; I wonder if it's possible to make the diagrams match those colour schemes instead of just the two above?

Thanks for the cool work, I am looking forward to seeing the pull request!

**Reply by IllustratedMan-code** [reply]
See https://github.com/aerickt/lapwing-for-beginners/pull/35. I managed to implement this + animations.

**Comment by evgenyfadeev** [comment]
@IllustratedMan-code it might or not help you but in the typeytype repo there is a React component that renders the steno keyboard.
It would need an adaptation to the use case that you're suggesting (it uses different input), but the presentational part is already there.

Perhaps the book could be migrated to Astro and there the react components could be integrated.

https://github.com/didoesdigital/typey-type

**Reply by IllustratedMan-code** [reply]
I managed to implement it fairly well in vanilla js I think. I'm certainly not a web developer, so dipping into react scares me.
